packages/mui-joy/src/CardOverflow/CardOverflow.tsx

```diff
@@ -44,67 +45,61 @@ const CardOverflowRoot = styled('div', {
   const childRadius = 'calc(var(--CardOverflow-radius) - var(--variant-borderWidth, 0px))';
   return {
     alignSelf: 'stretch', // prevent shrinking if parent's align-items is not initial
-    borderRadius: 'var(--CardOverflow-radius)',
     position: 'relative',
     display: 'flex',
-    ...(ownerState['data-parent'] === 'Card-horizontal' && {
+    margin: 'var(--_CardOverflow-margin)',
+    borderRadius: 'var(--_CardOverflow-radius)',
+    padding: 'var(--_CardOverflow-padding)',
```

Comment:
Without using internal CSS variables, the developer will need to override these properties with a selector to beat the `[`.${cardClasses.horizontal} > &`]` and `[data-first-child][data-last-child]`.